src/lib.rs

```diff
@@ -802,9 +803,11 @@ fn free_egui_textures_system(
         }
     }
 
-    for image_event in image_events.iter() {
-        if let AssetEvent::Removed { handle } = image_event {
-            egui_user_textures.remove_image(handle);
+    for image_event in image_events.read() {
+        if let AssetEvent::Removed { id } = image_event {
+            if let Some(id) = asset_server.get_id_handle(*id) {
+                egui_user_textures.remove_image(&id);
+            }
```

Comment:
I don't think this is working properly, I think it's already unloaded at this point so it would return none probably causing the egui texture to not be removed. I'm not sure what the proper solution would be yet since the event returns an AssetId instead of a handle so we can't look it up with the handle.